src/audio/multiband_drc/multiband_drc.c

```diff
@@ -113,75 +142,65 @@ static int multiband_drc_init_coef(struct processing_module *mod, int16_t nch, u
 
 	num_bands = config->num_bands;
 
-	/* Sanity checks */
-	if (nch > PLATFORM_MAX_CHANNELS) {
-		comp_err(dev,
-			 "multiband_drc_init_coef(), invalid channels count(%i)", nch);
-		return -EINVAL;
-	}
-	if (config->num_bands > SOF_MULTIBAND_DRC_MAX_BANDS) {
-		comp_err(dev, "multiband_drc_init_coef(), invalid bands count(%i)",
-			 config->num_bands);
+	if (nch > PLATFORM_MAX_CHANNELS || num_bands > SOF_MULTIBAND_DRC_MAX_BANDS) {
+		comp_err(dev, "Invalid ch count(%i) or band count(%i)", nch, num_bands);
 		return -EINVAL;
 	}
 
 	comp_info(dev, "multiband_drc_init_coef(), initializing %i-way crossover",
-		  config->num_bands);
+		  num_bands);
+
+	/* Allocation for coefficients_block */
+	if (!cd->coefficients_block) {
+		cd->coefficients_block = rballoc(0, SOF_MEM_CAPS_RAM,
```

Comment:
just allocate this upfront directly into `cd` so you don't have to constantly check this.

Reply:
Could you provide more detail on how to allocate `coefficients_block` directly with cd upfront, given that we typically need to check and allocate each separately in standard `C`?

Reply:
Done